analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToParameter_FromLiteral()
+        {
+            var setter = new LiteralDummyTestCheck();
+            var collector = SETestContext.CreateCS(@"boolParameter = true; Tag(""boolParameter"", boolParameter);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: boolParameter = true", x => x.State[((ISimpleAssignmentOperation)x.Operation.Instance).Target].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("boolParameter", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_CS()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                    return x.State.SetSymbolValue(x.Operation.Instance.TrackedSymbol(), x.State[x.Operation]);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"var b = boolParameter; Tag(""b"", b);", setter).Collector;
+            collector.ValidateTag("b", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_VB()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                    return x.State.SetSymbolValue(x.Operation.Instance.TrackedSymbol(), x.State[x.Operation]);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateVB(
+@"BoolParameter = True
+Dim B As Boolean = BoolParameter
+Tag(""B"", B)",
+setter).Collector;
+            collector.ValidateTag("B", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [DataTestMethod]
+        [DataRow("Sample.StaticField", "StaticField")]
+        [DataRow("Sample.StaticProperty", "StaticProperty")]
+        [DataRow("arr[0]", "array")]
+        [DataRow(@"dict[""key""]", "dictionary")]
+        [DataRow(@"other.Property", "other")]
+        [DataRow(@"this.Property", "this")]
+        public void SimpleAssignment_ToUnsupported_FromLiteral(string assignmentTarget, string tagName)
+        {
+            var setter = new LiteralDummyTestCheck();
+            var collector = SETestContext.CreateCS(
+                $@"{assignmentTarget} = 42; Tag(""{tagName}"", {assignmentTarget});",
```

Comment:
This doesn't need to be dynamic
```suggestion
                $@"{assignmentTarget} = 42; Tag(""Target"", {assignmentTarget});",
```